### PR TITLE
Update Readme

### PR DESCRIPTION
**Pull request by pizzalemon**
Updates README.md to have better information. Link to the readme and branch for your ease: https://github.com/pizzalemon/GroundStation/tree/readme-update.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,96 +1,127 @@
 # GroundStation
-Development Branch
 
-TJUAV's custom UI ground station
+TJUAV's custom ground station
```
Comment (krishnans2006) [inline README.md]:
```suggestion
TJUAV's custom Ground Station
```

File: README.md [hunk 2]
```diff
@@ -1,96 +1,127 @@
...
 
-## Virtual Environment
+## Interop
 
-### Create a virtualenv
+TJUAV's ground station interacts with the [Avusi Suas Interop Server](https://github.com/auvsi-suas/interop) as part of the Auvsi Suas competition. This is required, as of writing, for running the ground station.
```
Comment (krishnans2006) [inline README.md]:
```suggestion
TJUAV's Ground Station interacts with the [AUVSI SUAS Interop Server](https://github.com/auvsi-suas/interop) as part of the AUVSI SUAS competition. This is required, as of writing, for running the Ground Station. 
```

File: README.md [hunk 3]
```diff
@@ -1,96 +1,127 @@
...
+## Interop
 
-### Create a virtualenv
+TJUAV's ground station interacts with the [Avusi Suas Interop Server](https://github.com/auvsi-suas/interop) as part of the Auvsi Suas competition. This is required, as of writing, for running the ground station. 
 
-First make sure you have the virutalenv python library, run:
-`pip3 install virutalenv`
+### Setting Up Interop
+
+Clone the Auvsi Suas Interop server into the root directory of the ground station:
```
Comment (krishnans2006) [inline README.md]:
```suggestion
Clone the AUVSI SUAS Interop server into the root directory of the Ground Station: 
```

File: README.md [hunk 4]
```diff
@@ -1,96 +1,127 @@
...
 
-### Create a virtualenv
+TJUAV's ground station interacts with the [Avusi Suas Interop Server](https://github.com/auvsi-suas/interop) as part of the Auvsi Suas competition. This is required, as of writing, for running the ground station. 
 
-First make sure you have the virutalenv python library, run:
-`pip3 install virutalenv`
+### Setting Up Interop
+
+Clone the Auvsi Suas Interop server into the root directory of the ground station: 
+`git clone https://github.com/auvsi-suas/interop`. Next, navigate to the Auvsi Suas github page, read the introduction, and follow along to set up the Interop server.
```
Comment (krishnans2006) [inline README.md]:
```suggestion
`git clone https://github.com/auvsi-suas/interop`. Next, navigate to the AUVSI SUAS github page, read the introduction, and follow along to set up the Interop server.
```

File: README.md [hunk 5]
```diff
@@ -1,96 +1,127 @@
...
+
+The following is just a summary of the setup guide for Interop. Please still read the [Auvsi Suas Interop Github readme](https://github.com/auvsi-suas/interop), as it contains useful information about how each team's station will be set up during the actual competition. This section is just here for the benefit of those with prior experience of reading the readme. *However, there is still a section here which addresses errors in setting it up.*
```
Comment (krishnans2006) [inline README.md]:
```suggestion
The following is just a summary of the setup guide for Interop. Please still read the [AUVSI SUAS Interop Github README](https://github.com/auvsi-suas/interop), as it contains useful information about how each team's station will be set up during the actual competition. This section is just here for the benefit of those with prior experience of reading the README. *However, there is information below about some common errors and their fixes.*
```

File: README.md [hunk 6]
```diff
@@ -1,96 +1,127 @@
...
+
+You will want docker installed for this. You may want a basic understanding of Docker, as you may run into some errors. If you do not have Docker installed, search up the page and follows installation instructions, or download it from your linux package manager (you may need to install `docker-compose` as well).
```
Comment (krishnans2006) [inline README.md]:
```suggestion
You will want Docker installed for this. You may want a basic understanding of Docker, as you may run into some errors. If you do not have Docker installed, search up the page and follow installation instructions, or download it from your Linux package manager (you will need to install `docker-compose` as well).
```

File: README.md [hunk 7]
```diff
@@ -1,96 +1,127 @@
...
-in the outermost folder (GroundStation/)
+    sudo ./interop-server.sh up
 
-### Get into the venv
+We found this to be somewhat error prone, so you may want to try the following:
 
-For Linux run:
-`venv/bin/activate`
-and for Windows run:
+* Adding `user: postgres` to the `interop-db` section in `docker-compose.yml`
```
Comment (krishnans2006) [inline README.md:36]:
```suggestion
* Adding `user: postgres` to the `interop-db` section in `docker-compose.yml` (removes an error involving no root user present).
```

File: README.md [hunk 8]
```diff
@@ -1,96 +1,127 @@
...
+* Running the `create_db` command as *full* root. (Use `su` instead of `sudo`)
+* Is the healthcheck failing? It is likely that you ran `rm_data`, then ran `create_db`. 
+`rm_data` only removes the data. It does not delete the database. Run `drop_db`, then start the process over again.
+
+## Server
+
+The backend/server for the Groundstation is written in Python, with Flask as a web server for the frontend to communicate with.
```
Comment (krishnans2006) [inline README.md]:
```suggestion
The backend/server for the Ground Station is written in Python, with Flask as a web server for the frontend to communicate with.
```

File: README.md [hunk 9]
```diff
@@ -1,96 +1,127 @@
...
+
+### The Virtualenv
+
+First make sure you have the virutalenv python library, run:
```
Comment (krishnans2006) [inline README.md]:
```suggestion
First make sure you have the virtualenv python library, run:
```

File: README.md [hunk 10]
```diff
@@ -1,96 +1,127 @@
...
+`pip install virtualenv`. (Or download it from your distribution's package manager)
+
+Then, to create a virtualenv, run
+`python -m venv venv/`
+in the server folder `server/`
+
+To get into the venv, for Linux run
```
Comment (krishnans2006) [inline README.md]:
```suggestion
To get activate the virtual environment, for Linux run
```

File: README.md [hunk 11]
```diff
@@ -1,96 +1,127 @@
...
+Open a new terminal and run `python main.py` in `server/`.
+This starts the backend.
+If you make any edits, you will need to end the program and restart the backend.
 
-Run the above command in client/.
-Installs all of the necessary packages (specified in package-lock.json).<br />
+## Client
 
-### `npm i <package name>`
+The frontend/client for the Groundstation is written in React, along with some other libraries.
```
Comment (KrishnanS2006) [inline README.md:79]:
```suggestion
The frontend/client for the Groundstation is written in React, along with some other libraries.
The frontend is designed to repeatedly fetch information from the backend: therefore, it is not advisable to run the frontend before running the backend server. Doing so may result in an error or silent failure.
```

File: README.md [hunk 12]
```diff
@@ -1,96 +1,127 @@
...
-and for Windows run:
+* Adding `user: postgres` to the `interop-db` section in `docker-compose.yml`
+* Running the `create_db` command as *full* root. (Use `su` instead of `sudo`)
+* Is the healthcheck failing? It is likely that you ran `rm_data`, then ran `create_db`. 
+`rm_data` only removes the data. It does not delete the database. Run `drop_db`, then start the process over again.
+
+## Server
+
+The backend/server for the Groundstation is written in Python, with Flask as a web server for the frontend to communicate with.
+
```
Comment (krishnans2006) [inline README.md:45]:
```suggestion

The backend is designed to send and retrieve information from the interop server frequently: therefore, it is not advisable to run the backend before an interop server is online. Doing so may result in an error or loss of certain features.
```

File: README.md [hunk 13]
```diff
@@ -1,96 +1,127 @@
...
 
 ## Code Structure
 
 ### Frontend
 
 - Main code
-  - _index.js_: Contains the main code for the frontend.
-  - _App.js_: Contains the main code for the frontend.
+  - _index.js_: Frontend entry point
+  - _App.js_: Basic app structure; Renders routes/paths then each respective component
```
Comment (krishnans2006) [inline README.md]:
```suggestion
  - _App.js_: Basic app structure; Renders routes/paths and any components
```

File: README.md [hunk 14]
```diff
@@ -1,96 +1,127 @@
...
 - Tabs
-  - _FlightData.js_: Contains the plane's live telemetry data
-  - _FlightPlan.js_: Used to plan out waypoint missons.
+  - _FlightData.js_: Contains general plane data such as position or altitude, and some actions. Loads the Flight Map
   - _Params.js_: Contains a list of parameters that the user can edit and upload to the FC.
-  - _Submissions.js_: Displays all submissions that were made to the interop server.
+  - _Submissions.js_: ODLC submission page; Views all of the ODLCS from the backend, and it has the ability to edit them.
```
Comment (krishnans2006) [inline README.md]:
```suggestion
  - _Submissions.js_: ODLC submission page; Views all of the ODLCs from the backend, and has the ability to edit, submit, and/or reject them.
```

File: README.md [hunk 15]
```diff
@@ -1,96 +1,127 @@
...
-Generally, check out to a new git branch before beginning to work on a new change. Then, push all of these changes to that branch on the repo, and head over to the GitHub to create a new pull request. You can also fork the repository and merge it back in using a pull request after the changes are complete. When an entire feature is complete, create a pull request into the `master` branch, describe what changes are made, how they are implemented, or other notes, and add some other TJUAV programmers like [@Ganesh](https://github.com/gnanduru1), [@Jason Klein](https://github.com/Jklein64), or [@Srikar Gouru](https://github.com/srikarg89) to review it.
-
-### Note
+Create and check out a new git branch, branched from the `dev` branch, before beginning to work on a new change. When an entire feature is complete (don't open a PR for something that isn't done, unreviewed, or untested), push the branch to your fork, then open a PR to the `dev` branch; PR's should generally target `dev`. Describe what changes are made, how they are implemented, other notes, and add TJUAV programmers such as [@William](https://github.com/pizzalemon) or [@Krishnan](https://github.com/Jklein64) to review it.
```
Comment (krishnans2006) [inline README.md]:
```suggestion
Create and check out a new git branch, branched from the `dev` branch, before beginning to work on a new change. When an entire feature is complete (don't open a PR for something that isn't done, unreviewed, or untested, unless it is a Draft PR), push the branch to your fork, then open a PR to the `dev` branch; PR's should generally target `dev`. Describe what changes are made, how they are implemented, other notes, and add TJUAV programmers such as [@William](https://github.com/pizzalemon) or [@Krishnan](https://github.com/KrishnanS2006) to review it.
```

**Review by krishnans2006 — CHANGES_REQUESTED** [review]
This addresses a few typos, uncertainties, and other mistakes - these are very minor, and the rest is perfect for merge!

File: README.md (hunk 10 above)
Comment (krishnans2006) [reply README.md]: "get into" is very vague - maybe "activate" is a better term

File: README.md (hunk 9 above)
Comment (krishnans2006) [reply README.md]: This is a typo

File: README.md (hunk 12 above)
Comment (krishnans2006) [reply README.md:45]: Could you add this line?

File: README.md (hunk 8 above)
Comment (krishnans2006) [reply README.md]: You didn't change Ground Station in this part of the code

File: README.md (hunk 7 above)
Comment (krishnans2006) [reply README.md:36]: This is optional, but I feel like it would be better for understanding.

**Review by krishnans2006 — CHANGES_REQUESTED** [review]

**Review by krishnans2006 — APPROVED** [review]
Perfect - go ahead and merge it in!

**Pull request merged**